rerun_cpp/src/rerun/config.cpp

```diff
@@ -0,0 +1,37 @@
+#include "config.hpp"
+
+#include <algorithm>
+#include <cstdlib>
+#include <string>
+
+namespace rerun {
+
+    RerunGlobalConfig& RerunGlobalConfig::instance() {
+        static RerunGlobalConfig global;
+        return global;
+    }
+
+    RerunGlobalConfig::RerunGlobalConfig() {
+        // NOTE: we set the default value of default_enabled to match the environment
+        // variable, but in reality this doesn't actually do anything. The rust SDK
+        // will independently parse RERUN and disable `RecordingStream`s if it's set.
+        const char* envVarValue = std::getenv("RERUN");
+        if (envVarValue != nullptr) {
+            std::string envVarValueStr(envVarValue);
+            std::transform(
+                envVarValueStr.begin(),
+                envVarValueStr.end(),
+                envVarValueStr.begin(),
+                ::tolower
+            );
+            if (envVarValueStr == "1" || envVarValueStr == "true" || envVarValueStr == "yes") {
+                default_enabled.store(true, std::memory_order_seq_cst);
+            } else if (envVarValueStr == "0" || envVarValueStr == "false" || envVarValueStr == "no") {
+                default_enabled.store(false, std::memory_order_seq_cst);
+            }
+        } else {
+            default_enabled = true;
+        }
+    }
```

Comment:
I'm confused. If the whole point of the `RERUN` var is to override the value of `default_enabled`, then why are setting `default_enabled` to the value of the `RERUN` var...? What even is the point of `default_enabled` then?

I'd expect `default_enabled` to simply always be `true` unless the user changes it from code, like in other SDKs (I think?).

Reply:
I guess I can drop this -- there was some historical context where this whole config option was originally `set_enabled` as described in [the](https://github.com/rerun-io/rerun/issues/3754) and implemented before we remembered that rust was going to independently parse the RERUN var.